components/DetectAnimatedGif.coffee

```diff
@@ -22,7 +22,27 @@ exports.getComponent = ->
     async: true
     forwardGroups: true
   , (buffer, groups, out, callback) ->
-    out.send animated buffer
-    do callback
+    if Buffer.isBuffer buffer
+      out.send animated buffer
+      do callback
+      return
+    else if typeof buffer is 'string'
+      isAnimated = false
+      fs.createReadStream buffer
+        .pipe animated()
+        .once 'animated', ->
+          isAnimated = true
+          out.send true
+          do callback
+          return
+        .on 'finish', ->
+          unless isAnimated
```

Comment:
Why not send on finish, both in case that file is animated and case where it is not?
When you send affects when downstream components are triggered, so seems strange to have this vary...